include/deal.II/lac/linear_operator.h

```diff
@@ -1111,6 +1111,140 @@ linear_operator(const OperatorExemplar &operator_exemplar, const Matrix &matrix)
+ * For more details.
+ *
+ * @ingroup LAOperators
+ */
+template <  typename Range,
+            typename Domain,
+            typename Matrix >
+LinearOperator<Range, Domain>
+constraints_linear_operator(const ConstraintMatrix &cm, const Matrix &m)
+{
+  LinearOperator<Range, Domain> return_op = linear_operator<Range, Domain>(m);
+
+  return_op.vmult = [&cm](Range &v, const Domain &u)
+  {
+    for (auto i : v.locally_owned_elements())
+      {
+        if (cm.is_constrained(i))
+          {
+            v(i) = 0;
+            const std::vector< std::pair < types::global_dof_index, double > >
+            *entries = cm.get_constraint_entries (i);
+            for (types::global_dof_index j=0; j < entries->size(); ++j)
+              {
+                types::global_dof_index pos = (*entries)[j].first;
+                v(i) +=  u(pos) * (*entries)[j].second;
+              }
+          }
+        else
+          v(i) = u(i);
+      }
+  };
+
+  return_op.Tvmult = [&cm](Range &v, const Domain &u)
+  {
+    for (auto i : u.locally_owned_elements())
+      {
+        if (cm.is_constrained(i))
+          {
+            v(i)=0;
+            const std::vector< std::pair < types::global_dof_index, double > >
+            *entries = cm.get_constraint_entries (i);
+            for (types::global_dof_index j=0; j < entries->size(); ++j)
+              {
+                types::global_dof_index pos = (*entries)[j].first;
+                v(pos) += u(i) * (*entries)[j].second;
+              }
+          }
+        else
+          v(i)=u(i);
+
+      }
+  };
+
+  return return_op;
+}
+
+template <typename Range = Vector<double>,
+          typename Domain = Range,
+          typename Matrix>
+LinearOperator<Range, Domain>
+constrained_linear_operator(const ConstraintMatrix &, const Matrix &);
+
+/**
+ * @relates LinearOperator
+ *
+ * Let @p m be a system matrix and assume we are interested in this problem
+ * constrained with a constraint matrix @p cm.
+ * This function returns the LinearOperator that replace the system matrix and
+ * is associated to this problem.
+ * In detail, a constrained problem can be expressed in this form:
+ * Ct * S * C * x = Ct * ( rhs + S * k)
+ * Where:
+ *  - C has the role of constraint matrix
+ *  - k is the vector representing constraints inhomogeneities.
+ *    (a generic contraint can be expressed as x = Cx + k)
+ *  - S is the system matrix
+ *  - rhs is the original right-hand-side
+ * This function returns a LinearOperator representing the matrix Ct * S * C.
+ *
+ * @note Suppose we have n dof and m constraints. W.l.o.g. we can assume that it
+ * is possible to express n-m variables in terms of remainder variables
+ * (constrained variables).
+ * Therefore, x_i = C_{i,j} x_j + k_i for j = 1, ..., n-m and i = 1, ..., n
+ * Notice that Ct * S * C  is a problem in R^{n-m}, remaineder variables are
+ * treated solving x = 0 in order to have a well-posed problem on R^{n}.
```

Comment:
You need to put the `R^{n}` between dollar signs for this to be rendered as a formula. In that case, you can also write it as `${\mathbb R}^n$` to make it even prettier.